runtime/include/RuntimeCAPI.h

```diff
@@ -24,7 +24,7 @@ extern "C" {
 
 // Quantum Runtime Instructions
 void __catalyst__rt__fail_cstr(const char *);
-void __catalyst__rt__initialize();
+void __catalyst__rt__initialize(uint32_t *seed = nullptr);
```

Comment:
No defaults in C functions 😅 
```suggestion
void __catalyst__rt__initialize(uint32_t *);
```

Reply:
Oh this is CAPI, so no overloading and no default

So either we need to:
(a) update all tests to take in the pointer, or
(b) use a different function name

Reply:
Since using a different function name was vetoed some time before I will update the tests. 

Lee mentioned there might be more options we want to pass into rt_init in the future, so maybe a good idea is for it to take in a opaque struct pointer and let the struct carry the actual information; that won't happen in this PR but in that case updating the current tests seems like something that will happen anyway, so I'd just update them now.

Reply:
done

Reply:
Yes the issue is that adding the default in the header will work if you compile the source file with a C++ compiler, but would fail with a C compiler. But this header is supposed to be a strictly C-header 🤔 

> Lee mentioned there might be more options we want to pass into rt_init in the future, so maybe a good idea is for it to take in a opaque struct pointer and let the struct carry the actual information; that won't happen in this PR but in that case updating the current tests seems like something that will happen anyway, so I'd just update them now.

True I thought about the same thing where we could do a serialized dictionary with arbitrary config options but then I thought it would be overkill. As soon as we add a second option to the initialize function it would be a good idea though so that it's extensible without making changes to the signature 💯